tests/testevent.c

```diff
@@ -165,6 +165,107 @@ static LWES_BYTE ref_bytes_db[188] = {
   0xff,0xff,0xff,0x0b,0x61,0x4d,0x65,0x74,0x61,0x53,0x74,0x72,0x69,0x6e,0x67,
   0x05,0x00,0x05,0x68,0x65,0x6c,0x6c,0x6f};
 
+static int
+lwes_event_to_stream
+  (struct lwes_event *event,
+   FILE *stream)
+{
+  struct lwes_event_attribute *tmp;
+  struct lwes_hash_enumeration e;
+
+  fprintf (stream, "%s", event->eventName);
+  fprintf (stream,"[");
+  fflush (stream);
+  lwes_typed_value_to_stream (LWES_TYPE_U_INT_16, (void*)&event->number_of_attributes, stream);
+  fprintf (stream,"]");
+  fflush (stream);
+  fprintf (stream,"\n");
+  fflush (stream);
+  fprintf (stream,"{\n");
+  fflush (stream);
+
+  if (lwes_hash_keys (event->attributes, &e))
+    {
+      while (lwes_hash_enumeration_has_more_elements (&e))
+        {
+          LWES_SHORT_STRING tmpAttrName =
+            lwes_hash_enumeration_next_element (&e);
+
+          tmp =
+            (struct lwes_event_attribute *)lwes_hash_get (event->attributes,
+                                                          tmpAttrName);
+
+          fprintf (stream,"\t");
+          fflush (stream);
+          fprintf (stream, "%s", tmpAttrName);
+          fflush (stream);
+          fprintf (stream," = ");
+          fflush (stream);
+          lwes_event_attribute_to_stream (tmp, stream);
+          fflush (stream);
+          fprintf (stream,";\n");
+          fflush (stream);
+        }
+    }
+  fprintf (stream,"}\n");
+  fflush (stream);
+  return 0;
+}
+
+
+static LWES_BYTE array_event_bytes[] = {
+  14,65,114,114,97,121,84,101,115,116,69,118,101,110,116,0,13,3,101,110,
+  100,5,0,22,84,104,105,115,32,105,115,32,116,104,101,32,101,110,100,32,
+  102,105,101,108,100,46,9,115,116,114,105,110,103,78,65,114,145,0,8,0,8,
+  85,0,1,97,0,2,98,98,0,3,99,99,99,0,4,100,100,100,100,9,117,105,110,116,
+  49,54,78,65,114,141,0,14,0,14,171,40,0,1,0,2,0,3,0,5,0,7,0,11,0,13,8,
+  115,116,114,105,110,103,65,114,133,0,4,0,3,119,111,110,0,3,116,111,111,
+  0,4,70,114,101,101,0,3,102,111,114,8,117,105,110,116,49,54,65,114,129,
+  0,7,0,1,0,2,0,3,0,5,0,7,0,11,0,13,3,100,117,98,12,64,9,33,251,77,18,
+  216,74,8,102,108,111,97,116,105,110,103,11,102,254,244,249,4,98,105,
+  116,101,10,42,5,115,116,97,114,116,5,0,24,84,104,105,115,32,105,115,32,
+  116,104,101,32,115,116,97,114,116,32,102,105,101,108,100,46,14,120,95,
+  115,118,99,95,118,101,114,115,105,111,110,115,5,0,16,103,97,116,101,
+  119,97,121,47,49,49,46,49,49,55,46,48,10,120,95,97,112,112,95,110,97,
+  109,101,5,0,7,103,97,116,101,119,97,121,4,101,95,105,100,7,162,1,254,
+  124,103,80,96,101,9,101,95,118,101,114,115,105,111,110,4,0,1,96,24
+};
```

Comment:
I think it might be useful to get the serialized event from the erlang implementation https://github.com/lwes/lwes-erlang/blob/master/src/lwes_event.erl#L1230 and have another test which tests the deserialization of that.  Just as a double check that they are all interchangeable.

Reply:
Added around line 207.